stl/inc/vector

```diff
@@ -493,6 +493,10 @@ private:
             return;
         }
 
+        if (!is_ASan_enabled_for_allocator_v<allocator_type>) {

```

Comment:
Since this is testing a constant expression, it should probably be an `if constexpr` containing the entire function body so we don't bother to so much as compile it.

Reply:
`constexpr`'ed all the things: https://github.com/microsoft/STL/pull/5241/commits/71c7716a8a53bc36d97bad260f59662c3b1dc725

Reply:
I guess it's better to write

```C++
if constexpr (is_ASan_enabled_for_allocator_v<allocator_type>) {
// ASan stuffs
}
```

Reply:
Yeah I thought about that as well. In this case we're inside the `_Apply_annotation` method which, to my understanding, is already ASan-specific. I figure we wouldn't want to indent all of the remaining code to be inside the `if`-statement.

Do let me know if I'm misunderstanding something though :-)

Reply:
_Resolving due to inactivity. Please re-open if needed :-) _